### PR TITLE
A/B full screen tx test, with tab return

### LINKED ISSUE #7297: Full screen confirmation should still redirect to dapp after approval
I got the full screen confirmation window, and on completing the confirmation, was not redirected to the app, but instead the full screen account view, which was nice to see the pending transaction, but meant I totally lost dapp context.

### PR DESCRIPTION
**Pull request by danjm**
This PR fixes #7297 and turns back on the a/b test that was reverted with https://github.com/MetaMask/metamask-extension/pull/7298/files

The user will now be redirected back to a dapp after a full screen transaction

- [ ] still need to work through a couple edge cases

### CODE REVIEW

**Comment by Gudahtt** [comment]
We should consider reducing the percentage of users included in this A/B test. Something like... 10-20% might be better to start with maybe?

File: app/scripts/background.js [hunk 1]
```diff
@@ -394,7 +396,7 @@ function setupController (initState, initLangCode) {
         const origin = url.hostname
 
         remotePort.onMessage.addListener((msg) => {
-          if (msg.data && msg.data.method === 'eth_requestAccounts') {
+          if (msg.data && ['eth_requestAccounts', 'eth_sendTransaction'].includes(msg.data.method)) {
             requestAccountTabIds[origin] = tabId
```
Comment (Gudahtt) [inline app/scripts/background.js]: This stores the `tabId` for the _most recent_ tab to interact with MetaMask via either of these two methods. While this might be correct in most or all cases (at least so far), we should instead be able to find the `tabId` associated with specific requests instead, so that we can ensure we're redirecting the user back to the tab that initiated the specific confirmation they just finished.

**Comment by Gudahtt** [comment]
While I do think that this test is still worthwhile - there is an alternative I'd like to consider as well (injecting a "floating" confirmation page directly in the page, similar to the Firefox "Picture-in-picture" feature). Maybe it would be better to introduce these two tests at the same time? 20% full-screen, 20% floating, 60% left with the current behavior.

**Comment by Gudahtt** [comment]
Also since this A/B test relies upon metrics to provide useful feedback, these two metrics enhancements should come first before we turn this on: #7183 #8026

**Comment by whymarrh** [comment]
> two metrics enhancements should come first

Cool, I'll close this for now and we can reopen it once we're closer to merging
